knife/spec/unit/knife/bootstrap_spec.rb

```diff
@@ -1431,19 +1432,29 @@ def configure_gem_files
             knife.config[:connection_password] = "opscode"
           end
           it "includes :connection_password value in a sudo-enabled configuration" do
-            expect(knife.sudo_opts).to eq({
+            expect(knife.sudo_opts(sudo_pass)).to eq({
               sudo: true,
               sudo_password: "opscode",
             })
           end
+
+          context "when sudo_pass is passed" do
+            let(:sudo_pass) { "opscode" }
```

Comment:
Lets make this a different string so we know if we're taking sudo_pass or connection_password.
```suggestion
            let(:sudo_pass) { "progress" }
```